rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -1060,8 +1062,12 @@ public void onSuccess(String result) {
         }
     }
 
-    public void editNodeSource(String nodeSourceName) {
-        this.rmPage.showNodeSourceEditWindow(nodeSourceName);
+    public void editNodeSource(String nodeSourceName, NodeSourceStatus nodeSourceStatus) {
+        if (nodeSourceStatus.equals(NodeSourceStatus.NODES_UNDEPLOYED)) {
+            this.rmPage.showEditNodeSourceWindow(nodeSourceName);
+        } else {
+            this.rmPage.showEditDynamicParametersWindow(nodeSourceName);
+        }
     }
 
     /**
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)